src/Payments.sol

```diff
@@ -378,19 +370,46 @@ contract Payments is
             rail.from
         ][rail.operator];
 
-        // Ensure account lockup is fully settled up to the current epoch
+        // Settle account lockup as much as possible
         uint256 lockupSettledUpto = settleAccountLockup(payer);
+
+        // Check for rail in debt before modifying lockup
         require(
-            lockupSettledUpto == block.number,
-            "cannot modify active rail lockup: client funds insufficient for current account lockup settlement"
+            !isRailInDebt(rail, payer),
+            "cannot modify rail lockup: rail is in debt"
         );
 
-        // Calculate current (old) lockup - for active rails this is straightforward
+        // Only require full settlement if increasing period or fixed lockup
+        if (period > rail.lockupPeriod || lockupFixed > rail.lockupFixed) {
+            require(
+                lockupSettledUpto == block.number,
+                "cannot increase lockup: client funds insufficient for current account lockup settlement"
+            );
+        } else if (period < rail.lockupPeriod) {
+            // When reducing period, ensure we still cover all unsettled epochs
+            require(
+                payer.lockupLastSettledAt + period >= block.number,
+                "cannot reduce lockup period below what's needed for unsettled epochs"
+            );
```

Comment:
1. I think this check is redundant with `isRailInDebt`. I'd actually just keep this one and remove the `isRailInDebt` as, IMO, this is clearer. But up to you.
2. I don't think either check is actually necessary, but that's not what I wrote in the initial issue. If/when the rail is in debt, the effective lockup period is 0 so we won't unlock/relock any funds, but we can still reduce the period. I'm happy to leave this as-is for now.

Reply:
Issue filed as https://github.com/FilOzone/payments/issues/34.